### DISCUSSION TITLE
Testing subscriptions - shorter schedule

### DISCUSSION THREAD

**Discussion by vettloffah** [swellstores/community#138] [Help] [answered]
Right now, in order to test a subscription we have to wait a full day for the schedule to trigger the next order since the shortest possible schedule is 1 day.

Is there a way to either set a shorter schedule than one day (e.g. 5 minutes), or manually trigger the next subscription cycle?

As an aside - In Stripe, they offer a really nice feature called "test clocks" where you can advance the calendar manually to trigger events. This would be a nice feature in Swell.

**Comment by vettloffah** [comment, marked as answer]
Answering my own question here, support explained that you can go into the subscription in the dashboard, then click on where it says "Next order" and change the date in the UI.

<img width="621" alt="image" src="https://user-images.githubusercontent.com/3166892/218334540-e8fe5257-8a26-4fa2-bc97-b6310c21bca8.png">

**Comment by ericingram** [comment]
Test clocks sound interesting as I suppose it would affect the whole system instead of a single record.

You can also do it manually using the API:

```
PUT /subscriptions/<id>?date_order_cycle_start=<date in the past or near future>
```

Same goes for other cycle dates like `date_period_end` (used for billing vs fulfillment when they are different)
